### PR TITLE
ci: add commitlint github action

### PR DESCRIPTION
**Pull request by Madhu-1**
sometimes the commitlint GitHub app is not providing the proper results setting up a new GitHub action can help here.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

This is WIP need to update the mergify rules

### CODE REVIEW

File: .github/workflows/commitlint.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: wagoid/commitlint-github-action@v1.6.0
```
Comment (nixpanic) [inline .github/workflows/commitlint.yml:13]: Where did you find the docs for this? Please add that in the commit message or a comment in this file. I wonder what repository `wagoid` is...

**Comment by nixpanic** [comment]
Alternatively we can also run this in the CentOS CI

**Comment by nixpanic** [comment]
This is not available in the `Checks` tab yet. Is there something else needed to get this to work? Maybe a GitHub repository setting?

CI fails:
```
yamllint -s -d '{extends: default, rules: {line-length: {allow-non-breakable-inline-mappings: true}},ignore: charts/*/templates/*.yaml}' ./.github/workflows/commitlint.yml 
./.github/workflows/commitlint.yml
  1:1       warning  missing document start "---"  (document-start)
  2:1       warning  truthy value should be one of [false, true]  (truthy)
```

**Comment by Madhu-1** [comment]
> This is not available in the `Checks` tab yet. Is there something else needed to get this to work? Maybe a GitHub repository setting?
> 
> CI fails:
> 
> ```
> yamllint -s -d '{extends: default, rules: {line-length: {allow-non-breakable-inline-mappings: true}},ignore: charts/*/templates/*.yaml}' ./.github/workflows/commitlint.yml 
> ./.github/workflows/commitlint.yml
>   1:1       warning  missing document start "---"  (document-start)
>   2:1       warning  truthy value should be one of [false, true]  (truthy)
> ```

Its still a wip am working on it to enable

**Comment by Madhu-1** [comment]
I just did this on my local repo https://github.com/Madhu-1/test-github-actions/pull/1 actions seems to be shown in check. but not here

File: .github/workflows/commitlint.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+name: Commitlint
+on: [push, pull_request]  # yamllint disable-line rule:truthy
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (nixpanic) [inline .github/workflows/commitlint.yml:11]: this only checks out the last commit, you'll need a more complete history to check each commit

**Comment by nixpanic** [comment]
#1335 is an alternative. It adds a job in the CentOS CI which can override the `commitlint` status-context from the commitlint app. In case the app does not report the status, a comment with `/retest commitlint` will run the job in the CentOS CI.

**Comment by Madhu-1** [comment]
> #1335 is an alternative. It adds a job in the CentOS CI which can override the `commitlint` status-context from the commitlint app. In case the app does not report the status, a comment with `/retest commitlint` will run the job in the CentOS CI.

closing this one as per discussion on slack we can even add a comment to GitHub PR if something fails

**Pull request closed without merging**